SonarQube.Bootstrapper/Program.cs

```diff
@@ -40,12 +40,8 @@ private static int preprocess(ILogger logger, IBootstrapperSettings settings, st
         {
             string downloadBinPath = settings.DownloadDirectory;
 
-            logger.LogMessage(Resources.INFO_CreatingBinDir, downloadBinPath);
-            if (Directory.Exists(downloadBinPath))
-            {
-                Directory.Delete(downloadBinPath, true);
-            }
-            Directory.CreateDirectory(downloadBinPath);
+            Utilities.EnsureEmptyDirectory(settings.TempDirectory, logger);
```

Comment:
This is the line that does the work: the other changes are minor refactorings and clean-up.
I moved the existing EnsureEmptyDirectory method to SonarQube.Common so it could be called from this method too.